metrics/metrics_test.go

```diff
@@ -0,0 +1,52 @@
+package metrics
+
+import (
+	"testing"
+
+	"github.com/prometheus/client_golang/prometheus/testutil"
+	"github.com/stretchr/testify/assert"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+var defaultRegistry = metrics.Registry
+var find string = "odf_system_map"
+
+func TestReportODFSystemMapMetrics(t *testing.T) {
+	type args struct {
+		storageSystem, name, namespace, kind string
+	}
+	tests := []struct {
+		name string
+		args args
+	}{
+		{
+			name: "StorageSystem to StorageCluster",
+			args: args{
+				storageSystem: "StorageSystem1",
+				name:          "StorageCluster1",
+				namespace:     "Namespace1",
+				kind:          "StorageCluster",
+			},
+		},
+		{
+			name: "StorageSystem to Flashsystem",
+			args: args{
+				storageSystem: "StorageSystem2",
+				name:          "FlashSystem1",
+				namespace:     "Namespace1",
+				kind:          "Flashsystem",
+			},
+		},
+	}
+	for n, tt := range tests {
```

Comment:
Can you print the label here something like `t.Logf("Case %d: %s\n", i+1, tc.label)`

Reply:
t.Run() prints that for us.

Reply:
Correct. Here's a good rundown: https://pkg.go.dev/testing#hdr-Subtests_and_Sub_benchmarks

The output looks like this:
```
=== RUN   TestReportODFSystemMapMetrics
=== RUN   TestReportODFSystemMapMetrics/StorageSystem_to_StorageCluster
=== RUN   TestReportODFSystemMapMetrics/StorageSystem_to_Flashsystem
--- PASS: TestReportODFSystemMapMetrics (0.00s)
    --- PASS: TestReportODFSystemMapMetrics/StorageSystem_to_StorageCluster (0.00s)
    --- PASS: TestReportODFSystemMapMetrics/StorageSystem_to_Flashsystem (0.00s)
PASS
coverage: 100.0% of statements
ok  	github.com/red-hat-data-services/odf-operator/metrics	0.009s	coverage: 100.0% of statements
```